Structural and Performance improvements

balasankarc opened the pull request:
### General
- Use more sensible namespace: Follow `libindic.<module>.<language>` structure
- Migrate to testtools for testing
- Use pbr for packaging
- Add CI+Coverage for analysis and Tox for local automation.
### Malayalam
- Minimize false positives by using a root word corpus
- Handle multilevel inflections using iterative suffix stripping
- Add more tests

balasankarc committed:
Change to libindic.stemmer.language model

balasankarc committed:
Iterative suffix stripping with a rootword corpus

balasankarc committed:
Add tests

balasankarc committed:
Travis

balasankarc committed:
Use testtools

balasankarc committed:
Make travis run - Temporary

balasankarc committed:
Travis status on readme

balasankarc committed:
Use non-conflicting method name

balasankarc committed:
Merge branch 'multilevel' of https://github.com/balasankarc/indicstemmer into multilevel

balasankarc committed:
Fix directory structure

balasankarc committed:
Group the data files to a single folder

balasankarc committed:
Autopep8

balasankarc committed:
Remove backup file

balasankarc committed:
Use pbr, polish travis

balasankarc committed:
Flake8

balasankarc committed:
Add Coverage badge to readme

balasankarc committed:
Add normalizer to requirements

balasankarc committed:
Python3+Travis+Flake8

balasankarc committed:
Use custom normalizer

balasankarc committed:
Fix travis

balasankarc committed:
Coverage Badge

balasankarc committed:
Fix package name

balasankarc committed:
Merge branch 'multilevel' of github.com:balasankarc/indicstemmer into multilevel

balasankarc committed:
Try python 3.5-dev

balasankarc committed:
Configure coverage to exclude package-info functions

balasankarc committed:
Remove print commands used for development

balasankarc committed:
Fix coverage exclusion

balasankarc committed:
Use the original word given as input for dictionary keys

balasankarc committed:
Add tests for punctuation and existing root word

balasankarc committed:
Test for root word - Handle False positives

balasankarc committed:
Remove namespace package

balasankarc committed:
Ignore virtual environment directory

balasankarc committed:
Add tox configuration file

balasankarc committed:
Remove unnecessary namespace reference

balasankarc committed:
Use marisa_trie for storing root word corpus

Storing the root word corpus on a linear list is the least efficient way
to store it. Here, I switched it to use marisa_trie - a memory efficient
Trie datastructure based on marisa-trie C++ library. It is expected to
provide lesser space and time complexity.

balasankarc committed:
Switch to Markdown from ReStructuredText. :)

balasankarc committed:
Populate AUTHORS with git contributors

balasankarc committed:
Remove old README

balasankarc committed:
Fix documentation

balasankarc committed:
Cleanup Documentation

balasankarc committed:
Mimic existing Punjabi and Hindi stemmers: #TODO

balasankarc committed:
Point badges to the original repository

balasankarc committed:
Ignore Punjab and Hindi stemmers from coverage for now

balasankarc commented:
Go home coveralls, you are drunk. You got it in the wrong order. :smile:

copyninja commented on Makefile:
```diff
@@ -1,7 +1,7 @@
-		--coverage-package-name=stemmer
-	flake8 --max-complexity 10 libindic/stemmer
+		--coverage-package-name=libindic.stemmer
+	flake8 --max-complexity 15 libindic/stemmer
```
Why 15?. The value 10 is considered to be normal

balasankarc replied:
Ok. Malayalam.stem method had complexity of 10 and that was failing flake8 with max-complexity set to 10. Let me see if I can reduce that complexity and maintain max-complexity at 10. If it is not reducible, what shall I do?

BTW, existing Punjabi stemmer seems to have a complexity of 12, which will fail flake8 test. I don't intend to touch the Punjabi/Hindi stemmer for GSoC, as I am not familiar with their grammatical properties. Can I just delete those languages for now?

copyninja replied:
Just to give brief on complexity, it increases as you nest too much eg. conditionals. The way to reduce it refactor code such a way you don't nest too much. I've not checked the code so I can't give you exact reason but that is the idea.

I think it was added by @arushidogra lets see if she can help or not I will take a look later.

balasankarc replied:
Yeah. I've brought the complexity of Malayalam stemmer under 10.

copyninja replied:
Did you forget to push?. I didn't see the related comit yet

copyninja commented:
One general comment at many places comments doesn't make much sense like Travis Autopep8 and many places I note that commit message and commit content doesn't go together. Eg. Polish travis is commit message, commit contains changes to files under doc/ and many unrelated files.

This is not acceptable do atomic commits, i.e. commit only related files so that at any time we can go back and revert things as needed. For now I'm okay with this but if possible try to rebase and give proper commit messages than single word.

balasankarc commented:
Ok. I will keep that in mind and make atomic commits in future. Sorry for the trouble. :)

copyninja commented:
Good try to reduce complexity of your code and I shall merge the PR.

balasankarc committed:
Remove redundant conditional used in iteration

Remove use of 'completed' variable, which may be satisfied by the existing
'flag' variable. This helps in reducing complexity of the method and let
it pass the flake8 test.

balasankarc committed:
Set max-complexity of flake8 back to 10

copyninja commented:
OK I'm merging the request now

balasankarc commented:
@copyninja Thanks.

PS [OT]: BTW, is your name "Vasudev" or "Vasudeva" ? I see both of them intermixed in commits/mail notifications. :)

copyninja commented:
Err.. I did a mistake to use email id and name which was not supposed to be there. I've nuked the history sorry about it. Please pull carefully.

And phonetically its Vaasudeva but English is not phonetic ;-)

copyninja commented:
@balasankarc Please pull before you start working further!. I've renuked the history.

balasankarc commented:
@copyninja Ok.
